src/NumberField/NumberField.js

```diff
@@ -34,19 +34,53 @@ export default function NumberField({
     setNumber(value);
   }, [value]);
 
+  // helper function to round the number to the nearest step
+  const roundAccuratley = (number, decimalPlaces) => {
+    return Number(
+      Math.floor(number + "e" + decimalPlaces) + "e-" + decimalPlaces
+    );
+  };
+
+  // helper function to get the number of decimal places
+  const countDecimal = number => {
+    return number % 1 ? number.toString().split(".")[1].length : 0;
+  };
+
   // validate the number field when, number, min or max changes
   useEffect(() => {
     if (min !== undefined && number < min) {
+      // value is less than min
       setValueError(true);
       if (showMinMaxErrorMessage) {
-        setErrorMessage(`Must be greater than or equal to ${min}`);
+        setErrorMessage(`Must be greater than or equal to ${min}.`);
       }
     } else if (max !== undefined && number > max) {
+      // value is greater than max
       setValueError(true);
       if (showMinMaxErrorMessage) {
-        setErrorMessage(`Must be less than or equal to ${max}`);
+        setErrorMessage(`Must be less than or equal to ${max}.`);
       }
+    } else if (
+      number !== undefined &&
+      ((number * 100) % (step * 100)) / 100 !== 0
```

Comment:
Why do you need to multiply each term by 100?

Reply:
To avoid rounding error